source/server/listener_manager_impl.cc

```diff
@@ -767,6 +770,38 @@ uint64_t ListenerManagerImpl::numConnections() {
   return num_connections;
 }
 
+bool ListenerManagerImpl::stopListener(const std::string& name) {
```

Comment:
I'm wondering if we can simplify this change by just making this API take a `ListenerConfig`, and then just call the worker stop listener API directly with that config? I think this will work correctly.

I think no matter we have a potential race condition here in which if we remove anything, LDS will come and add it back later I think. Given this, it seems better to just stop the listeners directly? This would not cover the warming case, though that might be good enough? I think to completely remove any race conditions you would need to kill warming listeners and keep them from coming back by name? I'm really not sure if this type of complexity is worth it, but might be worth a bunch of comments.

WDYT?